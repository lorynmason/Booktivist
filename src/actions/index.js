export const addSearchResults = results => {
  return {
    type: 'ADD_RESULTS',
    results,
  }
}

export const addSearchInfo = info => {
  return {
    type: 'ADD_INFO',
    info,
  }
}

export const addMessage = message => {
  return {
    type: 'ADD_MESSAGE',
    message
  }
}