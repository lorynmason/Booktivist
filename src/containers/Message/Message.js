import React from 'react';
import { connect } from 'react-redux';
import { addMessage } from '../../actions';

export const Message = (props) => {
  setTimeout(() => {
    props.addMessage('')
  }, 5000);

  if(props.message) {
    return (
      <div className="message-container">
        <p>
          {props.message}
        </p>
      </div>
    )
  }
  return (
    <div className="blah">

    </div>
  )
}

export const mapStateToProps = (state) => ({
  message: state.message,
});

export const mapDispatchToProps = (dispatch) => ({
  addMessage: message => dispatch(addMessage(message))
});

export default connect(
  mapStateToProps, mapDispatchToProps
)(Message);