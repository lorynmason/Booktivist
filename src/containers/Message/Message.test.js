import React from 'react';
import { Message, mapStateToProps, mapDispatchToProps } from './Message'
import { shallow } from 'enzyme'
import { addMessage } from '../../actions'

describe('Message', () => {
  let wrapper;
  let message = 'There is a message'
  beforeEach(() => {
    wrapper = shallow(<Message addMessage={jest.fn()} message={message}/>)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should match snapshot if theres no message', () => {
    message = ''
    wrapper = shallow(<Message addMessage={jest.fn()} message={message}/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('mapStateToProps, should return a message', () => {
    const mockState = {
      results: [{ Name: 'Into the Wild' }, { Name: 'Looking for Alaska' }],
      message: '',
     bookList: [{ Name: 'Into the Wild' }, { Name: 'Looking for Alaska' }],
      isLoading: false
    };
    const expected = '';
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps.message).toEqual(expected);
  })

  it('mapDispatchToProps, should dispatch addMessage action when addMessage is called from props', () => {
    const mockDispatch = jest.fn();
    const mappedProps = mapDispatchToProps(mockDispatch)
    const expected = addMessage('You have a message');
    mappedProps.addMessage('You have a message')
    expect(mockDispatch).toHaveBeenCalledWith(expected)
  })
})