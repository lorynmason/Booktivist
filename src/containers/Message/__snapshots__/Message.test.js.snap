// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Message
    addMessage={[MockFunction]}
    message="There is a message"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        There is a message
      </p>,
      "className": "message-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "There is a message",
      },
      "ref": null,
      "rendered": "There is a message",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          There is a message
        </p>,
        "className": "message-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "There is a message",
        },
        "ref": null,
        "rendered": "There is a message",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Message should match snapshot if theres no message 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Message
    addMessage={[MockFunction]}
    message=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "blah",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "blah",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
